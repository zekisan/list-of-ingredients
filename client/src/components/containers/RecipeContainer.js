import React, { PureComponent } from 'react';
import '../../App.css';
import { Dimmer, Loader, Divider } from 'semantic-ui-react';

import TitlesToggle from '../../components/presentational/Home/TitlesToggle';
import Recipe from '../../components/presentational/Home/Recipe';

class Home extends PureComponent {
  constructor() {
    super();

    this.state = {}
    this.getDrinks = this.getDrinks.bind(this);
    this.getDrink = this.getDrink.bind(this);
  }

  componentDidMount() {
    this.getDrinks();
  }

  fetch(endpoint) {
    return window.fetch(endpoint)
      .then(response => response.json())
      .catch(error => console.log(error));
  }

  getDrinks() {
    this.fetch('/api/drinks')
      .then(drinks => {
        if (drinks.length) {
          this.setState({ drinks: drinks })
          this.getDrink(drinks[0].id)
        } else {
          this.setState({ drinks: [] })
        }
      })
  }

  getDrink(id) {
    this.fetch(`/api/drinks/${id}`)
      .then(drink => this.setState({ drink: drink }))
  }

  render() {
    let { drinks, drink } = this.state;
    return <span>
      {
        drinks
          ? <span>
            <TitlesToggle drinks={drinks} drink={drink} getDrink={this.getDrink} />
            <Divider section />
            <Recipe drink={drink} />
          </span>
          :
          <Dimmer active inverted>
            <Loader content='loading' />
          </Dimmer>
      }
    </span>
  }
}

export default Home;
