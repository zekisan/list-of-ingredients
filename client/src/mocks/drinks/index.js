const drinks = [
    {"id":1,"title":"Two-Minute Breakfast Boost","description":"Whizz up a low-fat breakfast smoothie in no time. Use banana with other soft fruit, plus honey for a little sweetness and oats for slow-release fuel.","steps":"Put all the ingredients in a blender and whizz for 1 min until smooth. Pour the mixture into two glasses to serve.","source":"https://www.bbcgoodfood.com/recipes/two-minute-breakfast-smoothie","created_at":"2018-06-09T00:22:22.524Z","updated_at":"2018-06-09T00:22:22.524Z","ingredients":[{"id":1,"description":"1 banana"},{"id":2,"description":"1 tbsp porridge oats"},{"id":3,"description":"80g soft fruit (like mango or strawberries)"},{"id":4,"description":"150ml milk"},{"id":5,"description":"1 tsp honey"},{"id":6,"description":"1 tsp vanilla extract"}]},
    {"id":2,"title":"Kale And Hearty Smoothie","description":"Give yourself a dose of vitamin C in the morning with this vegan green smoothie. Along with kale and avocado, there's a hit of zesty lime and pineapple.","steps":"Put all of the ingredients into a bullet or smoothie maker, add a large splash of water and blitz. Add more water until you have the desired consistency.","source":"https://www.bbcgoodfood.com/recipes/kale-smoothie","created_at":"2018-06-09T00:22:22.993Z","updated_at":"2018-06-09T00:22:22.993Z","ingredients":[{"id":7,"description":"2 handfuls kale"},{"id":8,"description":"½ avocado"},{"id":9,"description":"½ lime, juice only"},{"id":10,"description":"large handful frozen pineapple chunks"},{"id":11,"description":"medium-sized chunk ginger"},{"id":12,"description":"1 tbsp cashew nuts"},{"id":13,"description":"1 banana, optional"}]}
]

export default drinks;