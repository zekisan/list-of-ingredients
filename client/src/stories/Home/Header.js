import React from 'react';

import { storiesOf } from '@storybook/react';
import Header from '../../components/presentational/Home/Header';

storiesOf('Header', module).add('to Storybook', () => <Header />);