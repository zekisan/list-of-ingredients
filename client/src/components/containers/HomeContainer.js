import React, { Component } from 'react';
import '../../App.css';
import { Container, Dimmer, Loader, Divider } from 'semantic-ui-react';

import Header from '../../components/presentational/Home/Header';
import TitlesToggle from '../../components/presentational/Home/TitlesToggle';
import Recipe from '../../components/presentational/Home/Recipe';

class Home extends Component {
  constructor() {
    super();

    this.state = {}
    this.getDrinks = this.getDrinks.bind(this);
    this.getDrink = this.getDrink.bind(this);
  }

  componentDidMount() {
    this.getDrinks();
  }

  fetch(endpoint) {
    return window.fetch(endpoint)
      .then(response => response.json())
      .catch(error => console.log(error));
  }

  getDrinks() {
    this.fetch('/api/drinks')
      .then(drinks => {
        if (drinks.length) {
          this.setState({ drinks: drinks })
          this.getDrink(drinks[0].id)
        } else {
          this.setState({ drinks: [] })
        }
      })
  }

  getDrink(id) {
    this.fetch(`/api/drinks/${id}`)
      .then(drink => this.setState({ drink: drink }))
  }

  render() {
    let { drinks, drink } = this.state;
    return drinks
      ? <Container text>
        <Header />
        <Divider hidden section />
        <TitlesToggle drinks={drinks} drink={drink} getDrink={this.getDrink} />
        <Divider section />
        <Recipe drink={drink}/>
      </Container>
      : <Container text>
        <Dimmer active inverted>
          <Loader content='loadingsss' />
        </Dimmer>
      </Container>
  }
}

export default Home;
