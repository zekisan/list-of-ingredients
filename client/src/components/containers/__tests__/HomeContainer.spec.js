import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HomeContainer from '../HomeContainer';
import RecipeContainer from '../RecipeContainer';
import Header from '../../presentational/Home/Header';
import drinks from '../../../mocks/drinks'

describe('HomeContainer', () => {
    let mountedHomeContainer;
    const homeContainer = () => {
        if(!mountedHomeContainer) {
            mountedHomeContainer = mount(<HomeContainer />);
        }
        return mountedHomeContainer;    
    }

    beforeEach(() => {
        fetch.resetMocks()
        fetch.mockResponseOnce(JSON.stringify({ drinks: drinks }))

        mountedHomeContainer = undefined;
    })

    it('always renders a `Header`', () => {
        expect(homeContainer().find(Header).length).toBe(1);
    });

    it('always renders a `RecipeContainer`', () => {
        expect(homeContainer().find(RecipeContainer).length).toBe(1);
    });

    it('renders without crashing', () => {
        shallow(<HomeContainer />);
    });

    it('tries snapshots', () => {
        const tree = renderer
            .create(<HomeContainer />)
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });
})