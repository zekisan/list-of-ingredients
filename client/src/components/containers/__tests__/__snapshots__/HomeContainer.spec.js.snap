// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer tries snapshots 1`] = `
<div
  className="ui text container"
>
  <h2
    className="ui teal icon center aligned header"
  >
    <i
      aria-hidden="true"
      className="paper plane outline circular icon"
    />
    <div
      className="content"
    >
      List of Ingredients
    </div>
  </h2>
  <div
    className="ui hidden section divider"
  />
  <span>
    <div
      className="ui active transition visible inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui text loader"
        >
          loading
        </div>
      </div>
    </div>
  </span>
</div>
`;
